graphics/alert-graphic.html

```diff
@@ -6,20 +6,28 @@
 </head>
 <body>
     <div ng-app='alertsApp' ng-controller='alertsCtrl'>
-      <p ng-show='showAlert'>Good Play!</p>
+      <p ng-show='showAlertGoodPlay'>Good Play!</p>
+	    <p ng-show='showAlertBadLie'>Bad Lie!</p>
     </div>
     <script>
     var app = angular.module('alertsApp', []);
     app.controller('alertsCtrl', function($scope, $timeout) {
-      $scope.showAlert = false;
+      $scope.showAlertGoodPlay = false;
+      $scope.showAlertBadLie = false;
+      $scope.alertGoodPlay = function() {
+        $scope.showAlertGoodPlay = true;
+        $scope.$apply();
+        $timeout(function() {$scope.showAlertGoodPlay = false;}, 2000);
+      }
 
-      $scope.alert = function() {
-        $scope.showAlert = true;
+    $scope.alertBadLie = function() {
+        $scope.showAlertBadLie = true;
         $scope.$apply();
-        $timeout(function() {$scope.showAlert = false;}, 2000);
+        $timeout(function() {$scope.showAlertBadLie = false;}, 2000);
       }
 
-      nodecg.listenFor('goodPlay', $scope.alert);
+      nodecg.listenFor('goodPlay', $scope.alertGoodPlay);
+      nodecg.listenFor('badLie', $scope.alertBadLie);
```

Comment:
If you need help applying the abstraction to this file, let me know.  I have some ideas.

Reply:
I've spent a few hours trying to figure out THIS abstraction and I'm struggling. It seems like I'd need to send over an object with a property that contains the message content, but I'm not sure how to apply that to the 'listen for' directive. Hmmmm.....

Reply:
Hm... I'm actually realizing that this might be easier if we sent the messages differently.  sendMessage has the optional parameter of 'data'.  We could always send the alerts with the same messageName ('alert') but different data.

Each time we receive a message, the data is something like "Good Play!" or "Bad Lie!".  That string gets appended to an array.  Rather than using 'ng-show', we do ng-repeat on that array of messages, and each time we add a message, we set a time out to shift that array (removing the earliest entry in it).

```
$scope.displayedAlerts = [];

nodecg.listenFor('alert', $scope.alert);

$scope.alert = function(message) {
  $scope.displayedAlerts.push(message);
  $timeout(funtion() {$scope.displayedAlerts.shift()}, 2000);
}
```

Note that you may have to push `message.data` or something like that.  I'm not exactly sure.

Reply:
This model would mean that we could display multiple "Good Play!" alerts at once.  It might also make it hard to control how the transition of having the message appear and disappear.

So the other option would be to use a for loop on a similar object to the one that we used to create the buttons. Let's call that object `alerts`.  Something like:

```
$scope.showAlerts = {}

$scope.alert = function(message) {
  $scope.showAlerts[message.messageName] = true;
  $timeout(function() {$scope.showAlerts[message.messageName] = false}, 2000);
}

Object.keys(alerts).forEach(function(key) {
  nodecg.listenFor(key, alert);
}
```

And then there would have to be an ng-repeat in the html that created an element for each possible alert message in `alert` (so you might have to make a scope variable that is a copy of `alert`) and make it's ng-show attribute equal to `showAlerts[alert.message]`.